openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java

```diff
@@ -100,30 +104,95 @@ public OpenAPIConverterService() {
-            String fileUri = request.getDocumentFilePath();
-            Optional<SyntaxTree> syntaxTree = getPathFromURI(fileUri).flatMap(workspaceManager::syntaxTree);
-            Optional<SemanticModel> semanticModel = getPathFromURI(fileUri).flatMap(workspaceManager::semanticModel);
-            if (semanticModel.isEmpty() || syntaxTree.isEmpty()) {
-                StringBuilder errorString = getErrorMessage(syntaxTree, semanticModel);
-                response.setError(errorString.toString());
-            } else {
-                response.setError(null);
-                List<OASResult> oasResult = ServiceToOpenAPIConverterUtils.generateOAS3Definition(syntaxTree.get(),
-                        semanticModel.get(), null, false, null);
-                //Response handle with returning list of {@code OASResult} model.
-                response.setContent(oasResult);
+            Path filePath = Path.of(request.getDocumentFilePath());
+            Optional<SemanticModel> semanticModel = workspaceManager.semanticModel(filePath);
+            Optional<Module> module = workspaceManager.module(filePath);
+
+            if (semanticModel.isEmpty()) {
+                response.setError("Error while generating semantic model.");
+                return response;
+            }
+            if (module.isEmpty()) {
+                response.setError("Error while getting the module.");
+                return response;
             }
+            Module defaultModule = module.get();
+            JsonArray specs = new JsonArray();
+            for (DocumentId currentDocumentID : defaultModule.documentIds()) {
+                Document document = defaultModule.document(currentDocumentID);
+
+                SyntaxTree syntaxTree = document.syntaxTree();
+                List<OASResult> oasResults = ServiceToOpenAPIConverterUtils.generateOAS3Definition(
+                        syntaxTree, semanticModel.get(), null, false, null);
+
+                generateServiceJson(response, document.name(), oasResults, specs);
+            }
+            response.setContent(specs);
             return response;
         });
     }
 
+    /**
+     * Generate openAPI json for a service.
+     *
+     * @param response OpenAPIConverterResponse to set error message
+     * @param documentName Ballerina document name
+     * @param oasResults OAS Results list
+     * @param specs Specs array to add the service
+     */
+    private void generateServiceJson(OpenAPIConverterResponse response, String documentName,
+                                     List<OASResult> oasResults, JsonArray specs) {
+        for (OASResult oasResult : oasResults) {
+            if (oasResult.getOpenAPI().isPresent()) {
+                if (oasResult.getJson().isPresent()) {
+                    JsonObject spec = new JsonObject();
+                    JsonElement json = JsonParser.parseString(oasResult.getJson().get()).getAsJsonObject();
```

Comment:
Let's rename the `getJson()` method to a more meaningful name.

Reply:
Cannot rename because of the backward incompatibility.